src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py

```diff
@@ -3,8 +3,30 @@
 # License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------
 """Contains a class for generating VNF NFDs and associated resources."""
-from typing import Dict, Any
+import json
+import os
+import re
+import shutil
+import tarfile
+from typing import Dict, List, Any, Tuple, Optional, Iterator
+
+import yaml
 from azext_aosm.generate_nfd.nfd_generator_base import NFDGenerator
+from jinja2 import Template, StrictUndefined
+from knack.log import get_logger
+from azext_aosm._configuration import CNFConfiguration, HelmPackageConfig
+from azext_aosm.util.constants import (
+    CNF_DEFINITION_BICEP_TEMPLATE,
+    CNF_DEFINITION_JINJA2_SOURCE_TEMPLATE,
+    CNF_MANIFEST_BICEP_TEMPLATE,
+    CNF_MANIFEST_JINJA2_SOURCE_TEMPLATE,
+    DEPLOYMENT_PARAMETER_MAPPING_REGEX,
+    IMAGE_LINE_REGEX,
+    IMAGE_PULL_SECRET_LINE_REGEX,
+)
+from azure.cli.core.azclierror import InvalidTemplateError
+
+logger = get_logger(__name__)
```

Comment:
There is precious little logging in this class.  It is worth going through and thinking at each stage what you would want to know if you were debugging it, and write logs as such.  I'm going to leave you with this general comment rather than choose those log lines with individual comments myself.